### PR TITLE
update: pytorch_lightning and gpu training

### PR DESCRIPTION
**Pull request by georgosgeorgos**
* update trainers based on `lightning` to be compatible with `v1.7`
* add variable to handle `cudnn`

### CODE REVIEW

File: src/gt4sd/configuration.py [hunk 1]
```diff
@@ -50,6 +50,7 @@ class GT4SDConfiguration(BaseSettings):
     gt4sd_max_number_of_samples: int = 1000000
     gt4sd_max_runtime: int = 86400
     gt4sd_create_unverified_ssl_context: bool = False
+    gt4sd_disable_cudnn: bool = False
```
[the remark this answers is not in the document]
Comment (georgosgeorgos) [reply src/gt4sd/configuration.py:53]: Using it in `cli/trainer.py` with a conditional import.

**Review by drugilsberg — APPROVED** [review]
Worried a bit by the `pyarrow` upper bound but the rest looks very good. Let's see how the tests go.

**Comment by georgosgeorgos** [comment]
> Worried a bit by the `pyarrow` upper bound but the rest looks very good. Let's see how the tests go.

We had issues on the servers related to `aws-c-io` but we can relax the upper bound if needed.

https://github.com/huggingface/datasets/issues/3310

**Comment by drugilsberg** [comment]
> > Worried a bit by the `pyarrow` upper bound but the rest looks very good. Let's see how the tests go.
> 
> We had issues on the servers related to `aws-c-io` but we can relax the upper bound if needed.
> 
> [huggingface/datasets#3310](https://github.com/huggingface/datasets/issues/3310)

fine for the time being

**Pull request merged**